# include <rpm/rpmcrypto.h> in rpm-helper.h
# Reported upstream in https://github.com/OpenSCAP/openscap/pull/1922

From 989901ef95e344b49e78238c89c59bd80951b876 Mon Sep 17 00:00:00 2001
From: Tom Stellard <tstellar@redhat.com>
Date: Wed, 1 Feb 2023 06:38:40 +0000
Subject: [PATCH] Fix compile error with future versions of gcc

/builddir/build/BUILD/openscap-1.3.6/src/OVAL/probes/unix/linux/rpminfo_probe.c: In function 'rpminfo_probe_fini':
/builddir/build/BUILD/openscap-1.3.6/src/OVAL/probes/unix/linux/rpminfo_probe.c:307:9: error: implicit declaration of function 'rpmFreeCrypto'
  307 |         rpmFreeCrypto();
      |         ^~~~~~~~~~~~~
---
 src/OVAL/probes/unix/linux/rpm-helper.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/OVAL/probes/unix/linux/rpm-helper.h b/src/OVAL/probes/unix/linux/rpm-helper.h
index eca9cde04..6f7cb71d7 100644
--- a/src/OVAL/probes/unix/linux/rpm-helper.h
+++ b/src/OVAL/probes/unix/linux/rpm-helper.h
@@ -30,6 +30,7 @@
 #include <rpm/rpmts.h>
 #include <rpm/rpmmacro.h>
 #include <rpm/rpmlog.h>
+#include <rpm/rpmcrypto.h>
 #include <rpm/header.h>
 
 #include <pthread.h>
-- 
2.35.3

